packages/odyssey-react/src/components/NativeSelect/NativeSelectOptionGroup.tsx

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { forwardRef } from "react";
+import type { ComponentPropsWithRef, ReactNode } from "react";
+import { useOmit } from "../../utils";
+
+export interface NativeSelectOptionGroupProps
+  extends Omit<ComponentPropsWithRef<"optgroup">, "style" | "className"> {
+  /**
+   * Option children
+   */
+  children: ReactNode;
+}
+
+/**
+ * Native select input to choose one option from a list
+ */
+export const NativeSelectOptionGroup = forwardRef<
+  HTMLOptGroupElement,
+  NativeSelectOptionGroupProps
+>((props, ref) => {
+  const { children, ...rest } = props;
+  const omitProps = useOmit(rest);
+
+  return (
+    <optgroup {...omitProps} ref={ref}>
+      {children}
+    </optgroup>
+  );
+});
+
+NativeSelectOptionGroup.displayName = "NativeSelectOptionGroup";
```

Comment:
do we need a component for this? hmmm, it provides no styles today and no behavior but perhaps in the future it will?

Reply:
This mirrors the structure of Select and although it's not much more than a pass through the alternative feels odd in the context of all of our other component driven TSX. Just in this one place drop down into html...

```
<NativeSelect>
  <NativeSelect.OptionGroup>
     <NativeSelect.Option />
  </NativeSelect.OptionGroup>
</NativeSelect>

/* VS */

<NativeSelect>
  <optgroup>
     <option />
  </optgroup>
</NativeSelect>

```